### PR TITLE
aws - tagging - universal augment backwards compat

### PR DESCRIPTION
**Pull request by JohnHillegass**
Work in progress for https://github.com/cloud-custodian/cloud-custodian/issues/6515

### CODE REVIEW

File: c7n/tags.py [hunk 1]
```diff
@@ -82,15 +85,21 @@ def universal_augment(self, resources):
...
 
     rfetch = [r for r in resources if 'Tags' not in r]
-
     for arn_resource_set in utils.chunks(
             zip(self.get_arns(rfetch), rfetch), 100):
         arn_resource_map = dict(arn_resource_set)
-        resource_tag_results = client.get_resources(
-            ResourceARNList=list(arn_resource_map.keys())).get(
-                'ResourceTagMappingList', ())
+        if LooseVersion(botocore.__version__) < LooseVersion('1.19.58'):
```
Comment (JohnHillegass) [inline c7n/tags.py:95]: Need to add a test to cover this branch

Comment (JohnHillegass) [reply c7n/tags.py:95]: covered in 322554b

**Comment by kapilt** [comment]
i have mixed feelings on this, from an operator perspective this isnt one api call vs another .. its one api call vs o(n) aka bomb at scale. we've also since removed workarounds on using specific service apis wrt to tags. stronger guidance for a boto3 layer in the docs or auto layer seems a better workaround at least till lambda updates.

**Comment by JohnHillegass** [comment]
Fair enough! Forcing the lambdas to use the local c7n packages may be another approach. What do you think about editing mu for this? Something like `packages.extend(['boto3', 'botocore']`? This would always force the lambda to use what is current via local c7n dependencies.

**Comment by JohnHillegass** [comment]
Added some more comments to the issue to discuss proposed approaches to solving this one

**Comment by kapilt** [comment]
that significantly increases lambda size and volume usage as well execution times, for folks with thousands of lambda policies, that would also be problematic.. its really best if its a layer or we document the need in lambda policy execution as a best practice.

**Comment by JohnHillegass** [comment]
Hey @kapilt what do you think about some of the pros/cons listed in the issue? Theres a couple unknowns to me about how to implement this well in layers. 

**Option**
handle the version difference via #6521

Pros:

- operators don't need to worry about it
- as python runtime is updated, this will self deprecate
- there is one obvious way to handle this specific bug

Cons:

- O(n) on tagging lambdas until the python runtime is updated

**Option**
force an override via package upload in a layer

Pros:

- required packages can be toggled via a cli flag
- reuse of a layer helps to keep lambda size down
- there is one obvious way to handle this class of bugs

Cons:

- layer versioning and cleanup needs to be considered (increased management complexity)
- implications for layer version permission management (needs more investigation)
- indeterminate amount of time that operators need to supply the package overrides

**Comment by kapilt** [comment]
to be clear option 1, is not 
    O(n) on tagging lambdas until the python runtime is updated
its automatic issue on any lambda policy that has a resource using the work, we had specific workarounds per resource that have been removed around several high cardinality resources where this was an issue reported by users. this work is effectively a regression (rds snapshots as an example).

option 2 is viable longer term.. i don't think we even need to cli flag per se. not sure what your referencing about the last con here.

option 3, re more immediate 'fix' document the issue (ie recommend use of layers in lambda policies to avoid this class of issues), issue a warning on the cli to use an explicit layer if we detect resource group tag in a non config/poll policy.

**Comment by JohnHillegass** [comment]
Ok I'm starting to see what you are saying now. Will supersede this with something to add the auto layer feature.

**Comment by JohnHillegass** [comment]
Superceded by https://github.com/cloud-custodian/cloud-custodian/pull/6573

**Pull request closed without merging**